inference-engine/src/vpu/common/src/ngraph/operations/static_shape_broadcast.cpp

```diff
@@ -19,30 +19,32 @@ StaticShapeBroadcast::StaticShapeBroadcast(const Output<Node>& arg,
                                            const Output<Node>& targetShape,
                                            const Output<Node>& axesMapping,
                                            const ngraph::op::BroadcastModeSpec& broadcastSpec)
-        : ::ngraph::op::v3::Broadcast{arg, targetShape, axesMapping, broadcastSpec} {
+        : ::ngraph::op::v3::Broadcast{arg, targetShape, axesMapping, broadcastSpec},
+          m_evaluatedOutputShape{PartialShape::dynamic()} {
     constructor_validate_and_infer_types();
 }
 
 StaticShapeBroadcast::StaticShapeBroadcast(const Output<Node>& arg,
                                            const Output<Node>& targetShape,
                                            const ngraph::op::BroadcastModeSpec& broadcastSpec)
-        : ::ngraph::op::v3::Broadcast{arg, targetShape, broadcastSpec} {
+        : ::ngraph::op::v3::Broadcast{arg, targetShape, broadcastSpec},
+          m_evaluatedOutputShape{PartialShape::dynamic()} {
     constructor_validate_and_infer_types();
 }
 
 void StaticShapeBroadcast::validate_and_infer_types() {
-    if (get_output_partial_shape(0).is_static()) {
-        return;
-    }
-
     ::ngraph::op::v3::Broadcast::validate_and_infer_types();
```

Comment:
Does it work for cases when we eliminated `ShapeOf` operations (e.g. using `EliminateShapeOfAfterDSR`) and call `validate_and_infer_types` again?

Reply:
Yes, once the output shape is calculated, it will be stored as a class member and will be reused next time.